lib/metadata/models/object/object.js

```diff
@@ -0,0 +1,465 @@
+    getAmzVersionId() {
+        return this._data['x-amz-version-id'];
+    }
+
+    /**
+     * Set server version id
+     *
+     * @param {string} versionId - server version id
+     * @return {ObjectMD} itself
+     */
+    setAmzServerVersionId(versionId) {
+        this._data['x-amz-server-version-id'] = versionId;
+        return this;
+    }
+
+    /**
+     * Returns server version id
+     *
+     * @return {string} server version id
+     */
+    getAmzServerVersionId() {
+        return this._data['x-amz-server-version-id'];
+    }
+
+    /**
+     * Set storage class
+     *
+     * @param {string} storageClass - Storage class
+     * @return {ObjectMD} itself
+     */
+    setAmzStorageClass(storageClass) {
+        this._data['x-amz-storage-class'] = storageClass;
+        return this;
+    }
+
+    /**
+     * Returns storage class
+     *
+     * @return {string} Storage class
+     */
+    getAmzStorageClass() {
+        return this._data['x-amz-storage-class'];
+    }
+
+    /**
+     * Set server side encryption
+     *
+     * @param {string} serverSideEncryption - Server side encryption
+     * @return {ObjectMD} itself
+     */
+    setAmzServerSideEncryption(serverSideEncryption) {
+        this._data['x-amz-server-side-encryption'] = serverSideEncryption;
+        return this;
+    }
+
+    /**
+     * Returns server side encryption
+     *
+     * @return {string} server side encryption
+     */
+    getAmzServerSideEncryption() {
+        return this._data['x-amz-server-side-encryption'];
+    }
+
+    /**
+     * Set encryption key id
+     *
+     * @param {string} keyId - Encryption key id
+     * @return {ObjectMD} itself
+     */
+    setAmzEncryptionKeyId(keyId) {
+        this._data['x-amz-server-side-encryption-aws-kms-key-id'] = keyId;
+        return this;
+    }
+
+    /**
+     * Returns encryption key id
+     *
+     * @return {string} Encryption key id
+     */
+    getAmzEncryptionKeyId() {
+        return this._data['x-amz-server-side-encryption-aws-kms-key-id'];
+    }
+
+    /**
+     * Set encryption customer algorithm
+     *
+     * @param {string} algo - Encryption customer algorithm
+     * @return {ObjectMD} itself
+     */
+    setAmzEncryptionCustomerAlgorithm(algo) {
+        this._data['x-amz-server-side-encryption-customer-algorithm'] = algo;
+        return this;
+    }
+
+    /**
+     * Returns Encryption customer algorithm
+     *
+     * @return {string} Encryption customer algorithm
+     */
+    getAmzEncryptionCustomerAlgorithm() {
+        return this._data['x-amz-server-side-encryption-customer-algorithm'];
+    }
+
+    /**
+     * Set access control list
+     *
+     * @param {object} acl - Access control list
+     * @param {string} acl.Canned -
+     * @param {string[]} acl.FULL_CONTROL -
+     * @param {string[]} acl.WRITE_ACP -
+     * @param {string[]} acl.READ -
+     * @param {string[]} acl.READ_ACP -
+     * @return {ObjectMD} itself
+     */
+    setAcl(acl) {
```

Comment:
This needs a class to manage the ACL properties as well. Not in this PR, but definitely needs one.